lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -139,21 +159,96 @@ def _get(
             query = query.filter(Dataset.uploaded <= json["end"])
         if "name" in json:
             query = query.filter(Dataset.name.contains(json["name"]))
+        if "filter" in json:
```

Comment:
This additional code seems long enough to create separate method.  And then you could even just unit test that method.

What do you think?

Reply:
I'd considered breaking it out ... it's basically just a `transform Query -> Query`. May or may not make the code easier to read, and it would definitely be easier to test with mocks. Worth considering.

Reply:
I'm moving the filter query out, but I'm not going to break it up; it would be smaller, but I think harder to process in isolated chunks which would need a lot more context.